File: engine/model/src/main/scala/tools/packet/playerinteraction/LeaveHiredMerchant.scala
package tools.packet.playerinteraction

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class LeaveHiredMerchant(private var _slot: Int, private var _status: Int) extends PacketInput {
  def slot: Int = _slot

  def status: Int = _status

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/LeaveHiredMerchant.java
package tools.packet.character.interaction;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record LeaveHiredMerchant(Integer slot, Integer status) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}